docs/blog/2021-08-31-seeding-databases-tutorial.md

```diff
@@ -0,0 +1,733 @@
+> the [Prisma data model][prisma-schema] for a basic message board and how to
+> seed databases with [synth generated][getsynth] mock data to test our code.
+
+The code of the example we are working with here can be accessed in
+the [examples repository on GitHub][repo-complete].
+
+## Data modeling is not boring
+
+### What is a data model?
+
+![A sample data model (from Prisma's documentation)](./media/sample-database-model.png)
+
+Data modeling (in the context of databases and this tutorial) refers to the
+practice of formalizing a collection of entities, their properties and relations
+between one another. It is an almost mathematical process (borrowing a lot of
+language from [set theory][set-theory]) but that should not scare you. When it
+comes down to it, it is exceedingly simple and quickly becomes more of an art
+than a science.
+
+The crux of the problem of data modeling is to summarize and write down what
+constitutes useful entities and how they relate to one another in a graph of
+connections.
+
+You may wonder what constitutes a *useful* entity. It is indeed the toughest
+question to answer. It is very difficult to tackle it without a good combined
+idea of what you are building, the database you are building on top of and what
+the most common queries, operations and aggregate statistics are. There
+are [many resources][data-modeling-101] out there that will guide you through
+answering that question.
+
+### Why do I need a data model?
+
+Often times, getting the data model of your application right is crucial to its
+performance. A bad data model for your backend can mean it gets crippled by
+seemingly innocuous tasks. On the other hand, a good grasp on data modeling will
+make your life as a developer 1000 times easier. A good data model is not a
+source of constant pain, letting you develop and expand without slowing you
+down. It just is one of those things that pays out compounding returns.
+
+### Prisma is awesome
+
+[Prisma][prisma] is an ORM, an *object relational mapping*. It is a powerful
+framework that lets you specify your data model using a database agnostic domain
+specific language (called the [Prisma schema][prisma-schema]). It then
+uses [pluggable generators][prisma-generate] to code-generate a nice JS API and
+[typescript][typescript] bindings for your data model. Hook that up to your IDE
+and you get amazing code completion that is tailored to your data model.
+
+Let's walk through a example. We want to get a sense for what it'll take to
+design the data model for a simple message board a little like [Reddit][reddit]
+or [YCombinator's Hacker News][hacker-news]. We want to have users register for
+an account and write posts which can then be up voted or down voted by other
+registered users.
+
+Using the [Prisma schema][prisma-schema] language, which is very expressive even
+if you haven't seen it before, our first go at writing down a `User` entity
+might look something like this:
+
+```graphql
+model User {
```

Comment:
This definition is different to the actual Prisma file (I assume because you have since moved to MongoDB)